exec/cnex/cnex.c

```diff
@@ -331,8 +337,83 @@ void exec_raiseLiteral(TExecutor *self, TString *name, TString *info, Number cod
     }
     fprintf(stderr, "Unhandled exception %s (%s) (code %d)\n", TCSTR(name), TCSTR(info), number_to_sint32(code));
     cell_freeCell(exceptionvar);
-    // ToDo: Initiate a stack dump here...
+    cJSON *symbols = NULL;
+    cJSON *pStart = NULL;
+    while (self->ip < self->module->bytecode->codelen) {
+        char *dbgpath = strdup(self->module->source_path);
+        dbgpath[strlen(self->module->source_path)-1] = 'd';
+
+        FILE *fp = fopen(dbgpath, "rb");
+        // Don't report out of memory errors, let the stack unwinder report the best it can.
+        if (fp != NULL) {
+            fseek(fp, 0, SEEK_END);
+            long nSize = ftell(fp);
+            fseek(fp, 0, SEEK_SET);
+            char *debug = malloc(nSize);
+            if (debug != NULL) {
+                fread(debug, 1, nSize, fp);
+                fclose(fp);
+                free(dbgpath);
+                symbols = cJSON_Parse(debug);
+                // Remember the start of the symbols, so we can always go back to them.
+                pStart = symbols;
+                free(debug);
+            }
+        }
```

Comment:
Memory leak of `dbgpath` in the case where the file can't be found.

Reply:
Fixed.